Testscripts/Windows/VERIFY-XDP-COMPLIANCE.ps1

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+<#
+.Description
+    This script deploys the VM, build and Install xdpdump application
+    To verify xdp hook is working with current configuration.
+#>
+
+param([object] $AllVmData,
+	[object] $CurrentTestData)
+
+$SUPER_USER = "root"
+$MIN_KERNEL_VERSION = "5.6"
+
+function Main{
+    try{
+        Write-LogInfo "VM details:"
+        Write-LogInfo "  RoleName : $($allVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($allVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($allVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($allVMData.InternalIP)"
+
+        $currentKernelVersion = Run-LinuxCmd -ip $allVMData.PublicIP -port $allVMData.SSHPort `
+                -username $user -password $password -command "uname -r"
+        if ((Compare-KernelVersion $currentKernelVersion $MIN_KERNEL_VERSION) -lt 0){
+            Write-LogInfo "Unsupported kernel version: $currentKernelVersion"
+            return $global:ResultSkipped
+        }
+
+        # Provisioning VM
+        Provision-VMsForLisa -allVMData $allVMData -installPackagesOnRoleNames "none"
+
+        # Generate constants.sh and write all VM info into it
+        Write-LogInfo "Generating constants.sh"
+        $constantsFile = "$LogDir\constants.sh"
+        Set-Content -Value "# Generated by Azure Automation." -Path $constantsFile
+        Add-Content -Value "vms=$($allVMData.RoleName)" -Path $constantsFile
+        Add-Content -Value "ip=$($allVMData.InternalIP)" -Path $constantsFile
+        Add-Content -Value "nicName=eth1" -Path $constantsFile
+        foreach ($param in $currentTestData.TestParameters.param) {
+            Add-Content -Value "$param" -Path $constantsFile
+        }
+
+        Write-LogInfo "constants.sh created successfully..."
+        Write-LogInfo (Get-Content -Path $constantsFile)
+
+        #Build and Install XDP Dump application
+        $installXDPCommand = @"
+cd /root/
+./XDPDumpSetup.sh 2>&1 > /root/xdpConsoleLogs.txt
+. utils.sh
+collect_VM_properties
+"@
+        Set-Content "$LogDir\StartXDPSetup.sh" $installXDPCommand
+        Copy-RemoteFiles -uploadTo $allVMData.PublicIP -port $allVMData.SSHPort `
+            -files "$constantsFile,$LogDir\StartXDPSetup.sh" `
+            -username $SUPER_USER -password $password -upload
+
+        Run-LinuxCmd -ip $allVMData.PublicIP -port $allVMData.SSHPort `
+            -username $SUPER_USER -password $password -command "chmod +x *.sh" | Out-Null
+        $testJob = Run-LinuxCmd -ip $allVMData.PublicIP -port $allVMData.SSHPort `
+            -username $SUPER_USER -password $password -command "./StartXDPSetup.sh" `
+            -RunInBackground
+        while ((Get-Job -Id $testJob).State -eq "Running") {

```

Comment:
Need timer?

Reply:
You added the timer, but it does not seem to be best handle. If hitting 'break', how can you communicate out to the users?

Reply:
State.txt would not contain anything That's why it would go in else condition to "Aborted"

Reply:
It contains TestRunning, if I remember correct. If this is the case, this file may have TestCompleted or something else. Please review the flow of your logic. If you are condifent, then you can mark this conversation as 'resolved'

Reply:
> State.txt would not contain anything That's why it would go in else condition to "Aborted"

This is controlled by yourself in bash script. SetTestStateXXX